woocommerce/woocommerce-android
# Shipping rates networking

Closes: #12286 

### Description
This PR adds the network layer to connect shipping rates to the shipping labels' main flow. Besides its main goal of fetching shipping rates, I also reorganized the shipping rate classes into new packages (network, data source, and domain).
I also included a small improvement in this PR: sorting rates locally and removing the extra API request.

⚠️ I will add mapper classes unit tests in this PR https://github.com/woocommerce/woocommerce-android/pull/13138

### Testing information
1. Open the orders section
2. Select an order that is eligible to create shipping labels (processing state)
3. Tap on create shipping label
4. Select a package
5. Check that shipping rates are fetched from the API
6. Change the sorting order
7. Check that the shipping rates order is changed locally

### The tests that have been performed
1. Testing the UI works as expected
2. Testing the API response matches the rates displayed in the UI

### Images/gif

https://github.com/user-attachments/assets/5b6e6ebf-ce37-47db-a7b1-e19588ed590f




## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
